src/com/qmetry/qaf/automation/testng/RetryAnalyzer.java

```diff
@@ -68,13 +66,6 @@ public boolean retry(ITestResult result) {
 
 			getBundle().addProperty(RETRY_INVOCATION_COUNT, retryInvocationCount);
 
-			// correct failed invocation numbers for data driven test case.
-			List<Integer> failedInvocations = result.getMethod().getFailedInvocationNumbers();
-			if (null != failedInvocations && !failedInvocations.isEmpty()) {
-				int lastFailedIndex = failedInvocations.size() - 1;
-				failedInvocations.remove(lastFailedIndex);
-			}
-

```

Comment:
Why this is removed? This is especially to remove duplicate invocation numbers in generated testng-failed.xml for data driven tests.